<launch>
	<arg name="sub_instruction_topic_name"      default="/tb3omc/instruction" />
	<arg name="sub_depth_image_topic_name"      default="/camera/depth/image_raw" />
	<arg name="pub_joint_trajectory_topic_name" default="/tb3omc/joint_trajectory" />
	
	<arg name="sigverse_ros_bridge_port"        default="50001" />
	<arg name="ros_bridge_port"                 default="9090" />

	<arg name="camera"  default="camera" />

	<group ns="recognize_pointed_direction">
		<node name="recognize_pointed_direction" pkg="sigverse_turtlebot3_open_manipulator" type="recognize_pointed_direction" output="screen">

			<param name="sub_instruction_topic_name"      type="string" value="$(arg sub_instruction_topic_name)" />
			<param name="sub_depth_image_topic_name"      type="string" value="$(arg sub_depth_image_topic_name)" />
			<param name="pub_joint_trajectory_topic_name" type="string" value="$(arg pub_joint_trajectory_topic_name)" />

		</node>
	</group>

	<group ns="sigverse_ros_bridge">
		<node name="sigverse_ros_bridge" pkg="sigverse_ros_bridge" type="sigverse_ros_bridge" args="$(arg sigverse_ros_bridge_port)"/>
	</group>
	
	<include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" > 
		<arg name="port" value="$(arg ros_bridge_port)"/>
	</include>

	<include file="$(find realsense_camera)/launch/r200_nodelet_rgbd.launch">
		<arg name="camera"      value="$(arg camera)"/>
		<arg name="load_driver" value="false"/>
	</include>


	<node pkg="image_view" type="image_view" name="image_view" args="image:=$(arg camera)/depth/image_raw"/>

</launch>

